Games/LLEGSB/script.js

```diff
@@ -0,0 +1,269 @@
+  var ans1 = `x^2`;
+  var ans2 = `${rndmNum1+rndmNum2}x`;
+  var ans3 = `${rndmNum1*rndmNum2}`;
+  var finalAns = `${ans1}+${ans2}+${ans3}`;
+  correctAnswer = finalAns;
+}
+
+// question 9 function
+// introduction to the quadratic formula
+function question9(){
+  document.getElementById('demo').innerHTML = "Question 9";
+  confirmationText.text = '';
+  quitText.text = '';
+  hintText.text = '';
+  var a = Math.floor(Math.random() * (2 - 1 + 1)) + 1;
+  var b = Math.floor(Math.random() * (10 - 6 + 1)) + 6;
+  var c = Math.floor(Math.random() * (4 - 2 + 1)) + 2;
+  questionText.text = `Find the solutions(to 2 decimal places) of the equation ${a}x^2+${b}x+${c} using the quadratic formula. (Write as; #,# with the smallest number first.)`
+  const discriminant = (b**2-4*a*c);
+  const quadFormula1 = (-b+(discriminant)**0.5) / (2*a);
+  const quadFormula2 = (-b-(discriminant)**0.5) / (2*a);
+  var quad1Rounded = quadFormula1.toFixed(2);
+  var quad2Rounded = quadFormula2.toFixed(2);
+  correctAnswer = `${quad1Rounded},${quad2Rounded}`;
+}
+function checkAnswer(answer){
+  hintText.text = '';
+    if (answer == correctAnswer) {
+      confirmationText.text = 'Correct!';
+      playerPoints += 500;
+    }
+    else {
+      confirmationText.text = 'Wrong :(';
+    }
+}
+
+// function to change question on button push
+var currentQuestionIndex = 0;
+// list of functions
+const questionList = [question1, question2, question3, question4, question5, question6, question7, question8, question9]; 
+// calls next question
+function nextQuestion() {
+   questionList[currentQuestionIndex]();
+   currentQuestionIndex = (currentQuestionIndex + 1) % questionList.length; 
+}
+// hints
+const hint1 = "* Sum = Addition, Times = Multiply";
+const hint2 = "* x^2 = x*x";
+const hint3 = "* Multiply both inside #'s by the outside #";
+const hint4 = "* Combine the terms with the same letters";
+const hint5 = "* y2-y1...";
+const hint6 = "* Similar to Pythagorean Theorem";
+const hint7 = "* Any number to the power of 0 is 1";
+const hint8 = "* Use the FOIL method. Multiply the first terms of each binomial, the the outside terms, then the inside terms, then the last terms, and then combine like terms";
+const hint9 = "* The equation is in the form ax^2+bx+c. Knowing this, sub the a, b, and c values into the quadratic formula for your answer";
+// hint list
+const hintList = [hint1, hint2, hint3, hint4, hint5, hint6, hint7, hint8, hint9];
+// give player a hint at the cost of points
+function hintPlayer(){
+  hintText.text = hintList[currentQuestionIndex - 1];
+  confirmationText.text = '';
+  playerPoints -= 300;
+}
+// quit game on button press
+function quitGame(){
+  currentQuestionIndex = 0;
+  quitText.text = "Thanks for playing!";
+  pointsText.text = `You earned ${playerPoints} points!`
+  confirmationText.text = '';
+  questionText.text = '';
+  hintText.text = '';
+}
+//a timer to quit the game after 3 minutes
+setTimeout(quitGame, 180000)
```

Comment:
Numbers like this `180000` are often referred to as **magic numbers** because without documentation they tend to be mysterious, one-offs, and not reusable.  Magic numbers are a bad programming practice.  Assign the value to a named constant so that it is self-documenting and reusable.